conan/tools/meson/toolchain.py

```diff
@@ -71,8 +74,44 @@ def __init__(self, conanfile, env=os.environ):
         self._cppstd = cppstd_from_settings(self._conanfile.settings)
         self._shared = self._conanfile.options.get_safe("shared")
         self._fpic = self._conanfile.options.get_safe("fPIC")
+        self._build_env = VirtualEnv(self._conanfile).build_environment()
```

Comment:
I have merged this PR, because it is already an improvement.
But this is a bit "hardcoded". The ``VirtualEnv`` might be used in ``generate()`` in a different way, to customize what gets pulled into the environment, and users might prioritize dependencies, [conf], environment, or other stuff, but that will not be taken into account here.

It seems the ideal is that the MesonToolchain could use the ``conanbuildenv.xx`` script, or some other way to reuse the build environment. But I ignore how Meson works for this, I guess the activation of ``conanbuildenv.xx``  is still necessary?